### PR TITLE
Time to deprecate U and CX gates

### PR DESCRIPTION
**Pull request by 1ucian0**
This PR adds deprecation warnings when `U` and `CX` are used. Users should transition to `u3` and `cx`.

### CODE REVIEW

File: qiskit/extensions/standard/u3.py [hunk 1]
```diff
@@ -31,15 +29,6 @@ def __init__(self, theta, phi, lam, label=None):
         """Create new two-pulse single qubit gate."""
         super().__init__("u3", 1, [theta, phi, lam], label=label)
 
-    def _define(self):
```
Comment (lcapelluto) [inline qiskit/extensions/standard/u3.py]: I see `_define` used once in the codebase:
```
qiskit-terra/qiskit/circuit/instruction.py:
  158          """Return definition in terms of other basic gates."""
  159          if self._definition is None:
  160:             self._define()
  161          return self._definition
```
Could the `U3Gate` reach this? Where is `self._definition` set for this gate, perhaps through inheritance (or I just missed it)? Same questions for `CnotGate`

Comment (1ucian0) [reply qiskit/extensions/standard/u3.py]: Yes... it will be executed by inheritance. As before.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -35,6 +40,8 @@ Changed
 - When adding a register to a circuit, an error will now be raised if a register
   of the same name is already present. Previously, an error would only be raised
   if the same register was added twice.
+- Qubits and classical bits are not represented as a tuples anymore, but as
```
Comment (kdk) [inline CHANGELOG.rst]: This should be moved to the other PR.

Comment (1ucian0) [reply CHANGELOG.rst]: Fixed in 08a699a6

**Pull request merged**